### PR TITLE
Added loadout: US_Rangers_2020

### PR DESCRIPTION
**Pull request by AChesheireCat**
Adds a regiment of the 75th Rangers circa 2020.

### CODE REVIEW

File: addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp [hunk 1]
```diff
@@ -0,0 +1,657 @@
...
+class sl : r {
+	displayName = "Squad Leader";
+	vest[] = {
+		"CUP_V_CPC_tlbelt_mc",
+		"CUP_V_CPC_tl_mc"
+	};
+	backPack[] = {"CFP_Camelbak_Mule_Drab"};
+	sidearmWeapon[] = {"CUP_hgun_M17_Coyote"};
+	sidearmAttachments[] = {
+		"hlc_acc_tlr1",
```
Comment (Freddo3000) [inline addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp:187]: Missing dependency, NIArms Core (iirc)

File: addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp [hunk 2]
```diff
@@ -0,0 +1,657 @@
...
+	displayName = "Forward Air Controller";
+	goggles[] = {"antenna_long_two_face"};
+	vest[] = {"CUP_V_CPC_communicationsbelt_mc"};
+	backpack[] = {"B_Carryall_mcamo"};
+	backpackItems[] = {
+		FAC_GEAR ,
+		"ACRE_VHF30108"
+	};
+	magazines[] += {
+		"Laserbatteries",
```
Comment (Freddo3000) [inline addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp:221]: Should probably be in backpackItems, with the rest of the kit.

File: addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp [hunk 3]
```diff
@@ -0,0 +1,657 @@
...
+};
+
+class ar : r {
+	displayName = "Automatic Rifleman";
+	vest[] = {"CUP_V_CPC_lightbelt_mc"};
+	backpack[] = {"B_AssaultPack_mcamo"};
+	primaryWeapon[] = {"hlc_m249_pip4"};
+	scope[] = {"cup_optic_eotech553_black"};
+	attachment[] = {"cup_acc_anpeq_15"};
+	bipod[] = {"hlc_bipod_utgshooters"};
```
Comment (Freddo3000) [inline addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp:241]: Missing dependency?

File: addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp [hunk 4]
```diff
@@ -0,0 +1,657 @@
...
+	sidearmWeapon[] = {"CUP_hgun_M17_Coyote"};
+	sidearmAttachments[] = {
+		"hlc_acc_tlr1",
```
Comment (Freddo3000) [inline addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp:244]: Missing dependency?

File: addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp [hunk 5]
```diff
@@ -0,0 +1,657 @@
...
+	scope[] = {
+		"cup_optic_leupoldmk4",
+		"cup_optic_leupoldmk4_10x40_lrt_desert"
+	};
+	attachment[] = {"cup_acc_anpeq_15"};
+	silencer[] = {"cup_muzzle_snds_m110"};
+	bipod[] = {"cup_bipod_vltor_modpod"};
+	sidearmWeapon[] = {"CUP_hgun_M17_Coyote"};
+	sidearmAttachments[] = {
+		"hlc_acc_tlr1",
```
Comment (Freddo3000) [inline addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp:286]: Missing dependency?

File: addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp [hunk 6]
```diff
@@ -0,0 +1,657 @@
...
+};
+
+class mmgg : ar {
+	displayName = "MMG Gunner";
+	primaryWeapon[] = {"CUP_lmg_M240"};
+	scope[] = {"cup_optic_elcan_specterdr_black"};
+	backPack[] = {"CFP_Kitbag_MCam_Grn"};
+	magazines[] = {
+		"hlc_200Rnd_762x51_M_M60E4",
+		LIST_3("hlc_100Rnd_762x51_Barrier_M60E4"),
```
Comment (Freddo3000) [inline addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp:311]: Missing dependency, change to CUP belts.

File: addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp [hunk 7]
```diff
@@ -0,0 +1,657 @@
...
+		LIST_3("CUP_21Rnd_9x19_M17_Coyote")
+	};
+};
+
+class mmgac : aar {
+	displayName = "MMG Ammo Carrier";
+	backPack[] = {"CFP_Kitbag_MCam_Grn"};
+	backpackItems[] = {
+		LIST_4("hlc_100Rnd_762x51_Barrier_M60E4")
```
Comment (Freddo3000) [inline addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp:320]: Missing dependency

File: addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp [hunk 8]
```diff
@@ -0,0 +1,657 @@
...
+	};
+};
+
+class mmgtl : aar {
+	displayName = "MMG Team Leader";
+	items[] += {"ACE_MapTools"};
+	backPack[] = {"CFP_Kitbag_MCam_Grn"};
+	backpackItems[] = {
+		LIST_4("hlc_100Rnd_762x51_Barrier_M60E4")
```
Comment (Freddo3000) [inline addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp:329]: Missing dependency

File: addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp [hunk 9]
```diff
@@ -0,0 +1,657 @@
...
+	items[] += {"ACE_MapTools"};
+	linkedItems[] += {
+		"ACE_Vector"
+	};
+};
+
+class matg : car {
+	displayName = "MAT Gunner";
+	backpack[] = {"CFP_Kitbag_MCam_Grn"};
+	secondaryWeapon[] = {"CUP_launch_MAAWS"};
```
Comment (Freddo3000) [inline addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp:374]: Considering it is 2020, the vanilla MAAWS would probably fit nicely. https://en.wikipedia.org/wiki/Carl_Gustaf_recoilless_rifle#2014:_M4_(M3E1_in_USA)

File: addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp [hunk 10]
```diff
@@ -0,0 +1,657 @@
...
+	backPack[] = {"CFP_Camelbak_Mule_Drab"};
+	secondaryWeapon[] = {"CUP_launch_FIM92Stinger"};
+	linkedItems[] += {"Rangefinder"};
+};
+
+class sn : r {
+	displayName = "Sniper";
+	vest[] = {"CUP_V_CPC_light_mc"};
+	backpack[] = {"B_Carryall_mcamo"};
+	hmd[] = {"CUP_NVG_GPNVG_black"};
```
Comment (Freddo3000) [inline addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp:481]: No ghillies, or at least a boonie hat or such?

File: addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp [hunk 11]
```diff
@@ -0,0 +1,657 @@
...
+	silencer[] = {"cup_muzzle_snds_m16_coyote"};
+	sidearmWeapon[] = {"CUP_hgun_Glock17_blk"};
+	sidearmAttachments[] = {
+		"optic_mrd_black",
+		"hlc_acc_tlr1"
```
Comment (Freddo3000) [inline addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp:486]: Missing dependency

File: addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp [hunk 12]
```diff
@@ -0,0 +1,657 @@
...
+		LIST_4("CUP_20Rnd_762x51_B_M110"),
+		"ACE_SpottingScope",
+		"ACRE_VHF30108SPIKE"
+	};
+	linkedItems[] = {
+		"ItemMap",
+		"ItemCompass",
+		"ItemWatch",
+		"CUP_SOFLAM",
+		"B_UavTerminal"
```
Comment (Freddo3000) [inline addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp:519]: UAV terminal?

File: addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp [hunk 13]
```diff
@@ -0,0 +1,657 @@
...
+	backpack[] = {"B_Carryall_mcamo"};
+	sidearmWeapon[] = {"ACE_VMH3"};
+	items[] += {"ACE_Clacker"};
+	backpackItems[] = {
+		"ToolKit",
+		LIST_3("DemoCharge_Remote_Mag"),
+		LIST_2("SLAMDirectionalMine_Wire_Mag"),
+		"SatchelCharge_Remote_Mag",
+		"ACE_DefusalKit",
+		"ACE_EntrenchingTool",
```
Comment (Freddo3000) [inline addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp:628]: Wirecutters?

File: addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp [hunk 14]
```diff
@@ -0,0 +1,657 @@
...
+		"ACE_Clacker"
+	};
+	backpackItems[] = {
+		"ToolKit",
+		LIST_4("APERSBoundingMine_Range_Mag"),
+		LIST_2("APERSTripMine_Wire_Mag"),
+		LIST_2("ACE_FlareTripMine_Mag"),
+		LIST_2("ClaymoreDirectionalMine_Remote_Mag")
```
Comment (Freddo3000) [inline addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp:648]: Wirecutters?

**Review by Freddo3000 — CHANGES_REQUESTED** [review]
Considering we're possibly about to remove FIR, and that CUP pilot gear just got updated, I'd suggest you replace that.

File: addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp (hunk 10 above)
Comment (AChesheireCat) [reply addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp:481]: Nope, US Ranger Snipers don't run around in Ghillie suits. They run around in normal (but slick) kits - references: https://imgur.com/a/hXuuUTw

File: addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp (hunk 12 above)
Comment (AChesheireCat) [reply addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp:519]: To simulate the information network integration that F-35 and F-22 pilots have with other ISR assets within the area. Doubles as a GPS.

File: addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp (hunk 10 above)
Comment (AChesheireCat) [reply addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp:481]: chicken bullied me into giving them ghillies

File: addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp [hunk 15]
```diff
@@ -0,0 +1,670 @@
...
+		"CFP_Ranger_Beret_2Btn",
+		"CFP_Ranger_Beret_3Btn",
+		"CFP_75th_Patrol_Cap_Oakleys"
+	};
+	vest[] = {"CUP_V_CPC_light_mc"};
+};
+
+class fac : ftl {
+	displayName = "Forward Air Controller";
+	goggles[] = {"antenna_long_two_face"};
```
Comment (Freddo3000) [inline addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp:216]: Missing dependency

File: addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp [hunk 16]
```diff
@@ -0,0 +1,670 @@
...
+	linkedItems[] += {
+		"ACE_Vector"
+	};
+};
+
+class matg : car {
+	displayName = "MAT Gunner";
+	backpack[] = {"CFP_Kitbag_MCam_Grn"};
+	secondaryWeapon[] = {"launch_MRAWS_green_F"};
+	secondaryAttachments[] = {"cup_acc_anpeq_15"};
```
Comment (Freddo3000) [inline addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp:377]: Does the vanilla MAAWS take attachments?

File: addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp [hunk 17]
```diff
@@ -0,0 +1,670 @@
...
+class hc : smg {
+	displayName = "Helicopter Crew";
+	headgear[] = {
+		"CFP_SOARCREW_BLUE",
+		"CFP_SOARCREW_STACHE",
+		"CFP_SOARCREW_RED",
+		"CFP_SOARCREW_SKULL",
+		"H_CrewHelmetHeli_B",
+		"FIR_HGU56P_Skull",
+		"FIR_HGU56P_Skull_Death",
```
Comment (Freddo3000) [inline addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp:569]: See FIR issue

File: addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp [hunk 18]
```diff
@@ -0,0 +1,670 @@
...
+		"H_CrewHelmetHeli_O",
+		"H_PilotHelmetHeli_O"
+	};
+	uniform[] = {
+		"CUP_U_B_USArmy_ACU_Gloves_OEFCP",
+		"CUP_U_B_USArmy_ACU_OEFCP"
+	};
+	vest[] = {"FIR_pilot_vest"};
```
Comment (Freddo3000) [inline addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp:577]: See FIR issue

File: addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp [hunk 19]
```diff
@@ -0,0 +1,670 @@
...
+	backpackItems[] = {"ToolKit"};
+	linkedItems[] += {"ItemGPS"};
+};
+
+class jp : baseMan {
+	displayName = "Jet pilot";
+	hmd[] = {};
+	headgear[] = {"H_PilotHelmetFighter_B"};
+	uniform[] = {"FIR_Fighter_Pilot_Nomex3"};
+	vest[] = {"FIR_pilot_vest"};
```
Comment (Freddo3000) [inline addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp:610]: See FIR issue

File: addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp [hunk 20]
```diff
@@ -0,0 +1,670 @@
...
+	backpack[] = {"B_Parachute"};
+	primaryWeapon[] = {"CUP_arifle_mk18_black"};
+	scope[] = {};
+	attachment[] = {};
```
Comment (Freddo3000) [inline addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp:614]: Not necessary to be defined here, inheriting from unarmed.

**Review by Freddo3000 — CHANGES_REQUESTED** [review]

File: addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp (hunk 17 above)
Comment (AChesheireCat) [reply addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp:569]: LIES

File: addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp [hunk 21]
```diff
@@ -0,0 +1,670 @@
+// If you want to base a loadout on an existing one, this repository contains them all:
+// https://github.com/ARCOMM/arc_misc/tree/master/addons/tmf_loadouts
+
+//Dependencies:
+//		* CUP Uniforms
+//		* CUP Weapons
+//		* Community Factions Project
+//		* FIR AWS(AirWeaponSystem)
```
[the remark this answers is not in the document]
Comment (AChesheireCat) [reply addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp:8]: ninja says no u

File: addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp (hunk 16 above)
Comment (AChesheireCat) [reply addons/tmf_loadouts/loadouts/US_Rangers_2020.hpp:377]: Yes, funnily enough lol

I think it's only PEQ boxes tho

**Comment by Freddo3000** [comment]
On hold until FIR is clarified.

**Review by Freddo3000 — APPROVED** [review]

**Pull request merged**